IRSA auth for ECR doesn't work anymore

# Checklist

* [x] I've searched the issue queue to verify this is not a duplicate bug report.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `kargo version`.
* [x] I've pasted logs, if applicable.

# Description

After updating to 1.8.0 controller doesn't assume the IAM role anymore to generate token against ECR. The setup is working fine in 1.7.5, but stops working as soon as I update to 1.8.0. It works as soon as I downgrade to 1.7.5

While running 1.8.0 the IAM role shows that is had not been accessed and in controller logs I see that it's only trying to do a basic auth. ServiceAccount annotation is correct, IAM role trust relationship policy is correct, permissions to ECR for this role are correct.


# Version

```
Client Version: 1.8.0
Server Version: v1.8.0
```

# Logs

```
2025-10-21T16:20:47Z    DEBUG    warehouses/warehouses.go:140    reconciling Warehouse    {"namespace": "probe", "warehouse": "helm-charts"}
2025-10-21T16:20:47Z    DEBUG    warehouses/helm.go:49    found no credentials for chart repo    {"namespace": "probe", "warehouse": "helm-charts", "repoURL": "oci://1234567890.dkr.ecr.eu-central-1.amazonaws.com/foobar"}
2025-10-21T16:20:47Z    ERROR    warehouses/warehouses.go:160    error syncing Warehouse    {"namespace": "probe", "warehouse": "helm-charts", "error": "error discovering artifacts: error discovering charts: error discovering chart versions from helm chart repo \"oci://1234567890.dkr.ecr.eu-central-1.amazonaws.com/foobar\": error retrieving versions of chart from repository \"oci://1234567890.dkr.ecr.eu-central-1.amazonaws.com/foobar\": GET \"https://1234567890.dkr.ecr.eu-central-1.amazonaws.com/v2/foobar/tags/list\": basic credential not found"}
github.com/akuity/kargo/pkg/controller/warehouses.(*reconciler).Reconcile
    github.com/akuity/kargo/pkg/controller/warehouses/warehouses.go:160
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller[...]).Reconcile
    sigs.k8s.io/controller-runtime@v0.22.3/pkg/internal/controller/controller.go:216
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller[...]).reconcileHandler
    sigs.k8s.io/controller-runtime@v0.22.3/pkg/internal/controller/controller.go:461
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller[...]).processNextWorkItem
    sigs.k8s.io/controller-runtime@v0.22.3/pkg/internal/controller/controller.go:421
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller[...]).Start.func1.1
    sigs.k8s.io/controller-runtime@v0.22.3/pkg/internal/controller/controller.go:296
2025-10-21T16:20:47Z    DEBUG    warehouses/warehouses.go:180    done reconciling Warehouse    {"namespace": "probe", "warehouse": "helm-charts"}
2025-10-21T16:20:47Z    ERROR    controller/controller.go:421    Reconciler error    {"controller": "warehouse", "controllerGroup": "kargo.akuity.io", "controllerKind": "Warehouse", "Warehouse": {"name":"helm-charts","namespace":"probe"}, "namespace": "probe", "name": "he
lm-charts", "reconcileID": "b140031d-b9f2-46fc-a896-181b3105270a", "error": "error discovering artifacts: error discovering charts: error discovering chart versions from helm chart repo \"oci://1234567890.dkr.ecr.eu-central-1.amazonaws.com/foobar\": error
 retrieving versions of chart from repository \"oci://1234567890.dkr.ecr.eu-central-1.amazonaws.com/foobar\": GET \"https://1234567890.dkr.ecr.eu-central-1.amazonaws.com/v2/foobar/tags/list\": basic credential not found"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller[...]).processNextWorkItem
    sigs.k8s.io/controller-runtime@v0.22.3/pkg/internal/controller/controller.go:421
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller[...]).Start.func1.1
    sigs.k8s.io/controller-runtime@v0.22.3/pkg/internal/controller/controller.go:296
2025-10-21T16:20:47Z    DEBUG    warehouses/warehouses.go:140    reconciling Warehouse    {"namespace": "probe", "warehouse": "helm-charts"}
2025-10-21T16:20:47Z    DEBUG    warehouses/warehouses.go:340    freight creation criteria satisfied    {"namespace": "probe", "warehouse": "helm-charts"}
2025-10-21T16:20:48Z    DEBUG    warehouses/warehouses.go:180    done reconciling Warehouse    {"namespace": "probe", "warehouse": "helm-charts"}
```

--- a/pkg/credentials/kubernetes/ecr/access_key.go
+++ b/pkg/credentials/kubernetes/ecr/access_key.go
@@ -49,10 +49,6 @@ func (p *AccessKeyProvider) Supports(
 		return false
 	}
 
-	if credType == credentials.TypeHelm && !strings.HasPrefix(repoURL, "oci://") {
-		return false
-	}
-
 	if matches := ecrURLRegex.FindStringSubmatch(repoURL); len(matches) != 2 {
 		return false
 	}

--- a/pkg/credentials/kubernetes/ecr/access_key_test.go
+++ b/pkg/credentials/kubernetes/ecr/access_key_test.go
@@ -21,8 +21,7 @@ func TestNewAccessKeyProvider(t *testing.T) {
 
 func TestAccessKeyProvider_Supports(t *testing.T) {
 	const (
-		fakeRepoURL    = "123456789012.dkr.ecr.us-west-2.amazonaws.com/my-repo"
-		fakeOCIRepoURL = "oci://123456789012.dkr.ecr.us-west-2.amazonaws.com/my-repo"
+		fakeRepoURL = "123456789012.dkr.ecr.us-west-2.amazonaws.com/my-repo"
 
 		fakeRegion = "us-west-2"
 		fakeID     = "AKIAIOSFODNN7EXAMPLE"                     // nolint:gosec
@@ -50,7 +49,7 @@ func TestAccessKeyProvider_Supports(t *testing.T) {
 		{
 			name:     "valid helm oci credentials",
 			credType: credentials.TypeHelm,
-			repoURL:  fakeOCIRepoURL,
+			repoURL:  fakeRepoURL,
 			data: map[string][]byte{
 				regionKey: []byte(fakeRegion),
 				idKey:     []byte(fakeID),

--- a/pkg/credentials/kubernetes/ecr/common.go
+++ b/pkg/credentials/kubernetes/ecr/common.go
@@ -7,7 +7,7 @@ import (
 )
 
 // ecrURLRegex is a regex that matches ECR URLs.
-var ecrURLRegex = regexp.MustCompile(`^(?:oci://)?[0-9]{12}\.dkr\.ecr\.(.+)\.amazonaws\.com/`)
+var ecrURLRegex = regexp.MustCompile(`^[0-9]{12}\.dkr\.ecr\.(.+)\.amazonaws\.com/`)
 
 // tokenCacheKey returns a cache key in the form of a hash for the given parts.
 // Using a hash ensures that any sensitive data is not stored in a decodable

--- a/pkg/credentials/kubernetes/ecr/managed_identity.go
+++ b/pkg/credentials/kubernetes/ecr/managed_identity.go
@@ -6,7 +6,6 @@ import (
 	"fmt"
 	"net/http"
 	"os"
-	"strings"
 	"time"
 
 	"github.com/aws/aws-sdk-go-v2/aws"
@@ -82,7 +81,7 @@ func NewManagedIdentityProvider(ctx context.Context) credentials.Provider {
 
 func (p *ManagedIdentityProvider) Supports(
 	credType credentials.Type,
-	repoURL string,
+	_ string,
 	_ map[string][]byte,
 	_ map[string]string,
 ) bool {
@@ -91,10 +90,6 @@ func (p *ManagedIdentityProvider) Supports(
 	}
 
 	if credType != credentials.TypeImage && credType != credentials.TypeHelm {
-		return false
-	}
-
-	if credType == credentials.TypeHelm && !strings.HasPrefix(repoURL, "oci://") {
 		return false
 	}
 

--- a/pkg/credentials/kubernetes/ecr/managed_identity_test.go
+++ b/pkg/credentials/kubernetes/ecr/managed_identity_test.go
@@ -14,9 +14,8 @@ import (
 
 func TestManagedIdentityProvider_Supports(t *testing.T) {
 	const (
-		fakeAccountID  = "123456789012"
-		fakeRepoURL    = "123456789012.dkr.ecr.us-west-2.amazonaws.com/my-repo"
-		fakeOCIRepoURL = "oci://123456789012.dkr.ecr.us-west-2.amazonaws.com/my-repo"
+		fakeAccountID = "123456789012"
+		fakeRepoURL   = "123456789012.dkr.ecr.us-west-2.amazonaws.com/my-repo"
 	)
 
 	testCases := []struct {
@@ -45,22 +44,13 @@ func TestManagedIdentityProvider_Supports(t *testing.T) {
 			expected: true,
 		},
 		{
-			name: "OCI helm credentials supported",
+			name: "helm credentials supported",
 			provider: &ManagedIdentityProvider{
 				accountID: fakeAccountID,
 			},
 			credType: credentials.TypeHelm,
-			repoURL:  fakeOCIRepoURL,
+			repoURL:  fakeRepoURL,
 			expected: true,
-		},
-		{
-			name: "non-OCI helm credentials not supported",
-			provider: &ManagedIdentityProvider{
-				accountID: fakeAccountID,
-			},
-			credType: credentials.TypeHelm,
-			repoURL:  "https://123456789012.dkr.ecr.us-west-2.amazonaws.com/repo",
-			expected: false,
 		},
 		{
 			name: "git credentials not supported",

